packages/@lwc/template-compiler/src/parser/expression.ts

```diff
@@ -107,9 +107,13 @@ function validateSourceIsParsedExpression(source: string, parsedExpression: Node
 // FIXME: Avoid throwing errors and return it properly
 export function parseExpression(source: string, element: IRNode, state: State): TemplateExpression {
```

Comment:
This PR makes me really think that we should revise our approach on how we parse expressions. We are currently facing these issues because we are using a regex to extract expression out of text nodes. I am wondering an approach where we traverse the text until reaching `{` and start parsing the expression from there would help.